test/utils/node-http.js

```diff
@@ -0,0 +1,19 @@
+const util = require('node:util')
+
+function closeServerAsPromise (server) {
+  return () => util.promisify(server.close.bind(server))()
+}
+
+function closeClientAndServerAsPromise (client, server) {
```

Comment:
When testing http2 server/client, all sessions must be closed before closing the server. ref: https://nodejs.org/api/http2.html#serverclosecallback_1